core/src/message/init.rs

```diff
@@ -178,6 +178,23 @@ impl InitPacket {
         }
     }
 
+    /// Create new InitPacket with optional gas.
+    pub fn maybe_with_gas(
```

Comment:
If you want to also do #3320, we need 3 functions:
No message, with gas (new_from_user)
With message, no gas (new_from_program)
With message, with gas (new_from_program_with_gas)

It shouldn't be possible to create an InitPacket from user (no message id) without gas.

This would save me opening a new PR just for this simple rename/refactor of functions

Reply:
Looking at the way you expanded the macros, it might be better to just keep two function:
new_from_user: no messageId, with gas (gas is mandatory)
new_from_program: messageId (mandatory), Option<gas>

Reply:
Done